### DISCUSSION TITLE
Enable SN Multi Language support

### DISCUSSION THREAD

**Discussion by uncle-jj** [stackernews/stacker.news#1875] [Ideas]
**Is your feature request related to a problem? Please describe.**
Not everyone reads (and is comfortable writing) in English.

**Describe the solution you'd like**
SN need to expand its capabilities to onboard anyone able to type on a keyboard. Not sure how it could be done. I think about Ai or third parties APIs translation services that could facilitate and solve the workload in real time. The system should also detect the browser default language and provide all the contents translated into a that specific languages.

**Describe alternatives you've considered**
- Alice will be able to post and real all contents in Italian
- Bob will be able to read Alice posts and respond with a comment in English 
- Carlos will be able to post and read Alice and Bob posts in Spanish and comment in Spanish
- BoB will be able to read Carlos comment in English and respond in English
- Alice will be able to read Carlos and Bob comments in Italian and respond in Italian
... 

**Additional context**
Either, if browser language is detected, the user should be able to select its preferred language in the setting page.

**Comment by Darth-Coin** [comment]
Bitcoin universal language is English. We like it or not, this is it!
If you want a separate language, just create a SN territory in that language or learn English FFS.

**Reply by adlai** [reply]
Sorry, Darth, you're simply wrong; Bitcoin is not universal, it is planetary.

The nascent language of our planetary coin was English; that might not last beyond the first few nonflation epochs [ nonflation epoch = how many years would it have taken the [BIP0042](https://github.com/bitcoin/bips/blob/master/bip-0042.mediawiki#details) rollover undefined behavior to recur, first one projected to occur around 2139 ]

**Comment by uncle-jj** [comment]
Referring @huumn [comment](https://stacker.news/items/219228/r/mo?commentId=219413)

Minimum viable PR
- plug DeepL API or implement any other LLM to automate translations
- set a preferred language in your settings
- add a translate button to any post or comment text that isn't in that language (it could be added as option in the menu)

![Image_01](https://github.com/stackernews/stacker.news/assets/567747/40f0b99f-4f99-45f8-80ad-6d8478c86edc)
![Image_02](https://github.com/stackernews/stacker.news/assets/567747/84f4281c-b3a4-4003-92a1-4f0b7f4b9471)
![Image_03](https://github.com/stackernews/stacker.news/assets/567747/d3199f97-a806-4085-8915-4b3adcef7d53)
![Image_04](https://github.com/stackernews/stacker.news/assets/567747/de5aa85d-dfbd-4974-88b3-6acff1797e99)


Also, insteat the maual {right-click+traslate} SN could monetize from offering the service to enhance the experience. For example, enabling a **translate everything** option (it will have a `x` sats per request - also need a log to record all translation request in case are needed from the user)

![Image_05](https://github.com/stackernews/stacker.news/assets/567747/cb71a7c9-53c9-4275-b7f3-a7857adb0848)
![Image_06](https://github.com/stackernews/stacker.news/assets/567747/4835f0cd-38b3-4d01-8d9f-c98d9db36174)

**Reply by adlai** [reply]
worst idea, ever

you wanna "fix online communities with economics"?

kill duolingo ; pay interpreters.

**Comment by ekzyis** [comment]
Just fyi, I wrote a tampermonkey script which adds translate buttons here: https://git.ekzyis.com/ekzyis/sn-translator

It doesn't work anymore iirc (I think I hit rate limits) but it can be seen as a rough draft how this could be dealt with solely on the client.

**Comment by uncle-jj** [comment]
Thanks @ekzyis it's a great solution from the user/client perspective, and the market also offers alternatives to this too. My aim was more to highlight the benefits it could bring, mostly to SN.

**Comment by adlai** [comment]
I almost caused physical damage to my thin client after reading one too many comments dat rite like dis; shud it be konsidered "a you problem" ?
